gramex/data.py

```diff
@@ -1381,9 +1382,16 @@ def _type_conversion(param_list, operations=None):
 }
 
 
-def _filter_mongodb_col(col, op, vals, meta_cols):
+def _filter_mongodb_col(col, op, vals, meta_cols, object_key=None):
+    import bson
+
     if op in ['', '!']:
-        return {col: {_mongodb_op_map[op]: vals}}
+        convert = int if (meta_cols[col].dtype == pd.np.int64) else \
```

Comment:
This conversion needs to apply even if the value is `>` or `<` etc. Convert the type for all values based on the type.